data/framer.go

```diff
@@ -0,0 +1,8 @@
+package data
+
+// Framer is simply an object that can be converted to Grafana data frames.
+// This interface allows us to interact with types that represent datasource objects
+// Without having to convert them to dataframes first
+type Framer interface {
+	Frames() Frames
```

Comment:
I think this should probably be able to return an error. Structs may have properties etc that can fail in Frame conversion.

Also keep in mind that once we add an interface, we can not add methods to its without making a new major version of the SDK (we can create new interfaces, but a new method on an interface is breaking change.

Reply:
hmm. Yeah I didn't encounter any error cases when working with the GitHub plugin but since that was GraphQL I exercised quite a bit of control over what I could query. I could see this having to return an error in something like the ServiceNow data source.

I'll add an error argument to this.

Reply:
re: "Also keep in mind that once we add an interface, we can not add methods to its without making a new major version of the SDK"

Good to know! I don't foresee any need to add additional functions to this interface. I think it kind of loses its magic if you have to implement more than one function.

The only thing I can think of is maybe changing the signature to include a `context.Context` but I don't think that should be necessary.